### PR TITLE
ARROW-4472: [Website][Python] Blog post about string memory use work in Arrow 0.12

### PR DESCRIPTION
**Pull request by wesm**
This blog shows how we were able to significant improve performance and memory use in common cases when converting from the Arrow string memory layout to pandas's native memory model based on NumPy arrays of Python objects.

### CODE REVIEW

**Comment by wesm** [comment]
Here's a published version https://wesm.github.io/arrow-site-test/blog/2019/02/04/python-string-memory-0.12/

test publishing the website really isn't that easy, and a number of things are broken...

**Comment by fsaintjacques** [comment]
LGTM

File: site/_posts/2019-02-05-python-string-memory-0.12.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+faster in addition to being vastly more memory efficient in the common case in
+analytics where we have table columns with many instances of the same string
+values.
+
+## Memory and Performance Benchmarks
+
+We can use the `memory_profiler` Python package to easily get process memory
+usage within a running Python application.
+
+```
```
Comment (xhochy) [inline site/_posts/2019-02-05-python-string-memory-0.12.md]:
```suggestion
```python
```

File: site/_posts/2019-02-05-python-string-memory-0.12.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+In `pyarrow` 0.12, we have implemented this when calling `to_pandas`. It
+requires using a hash table to deduplicate the Arrow string data as it's being
+converted to pandas. Hashing data is not free, but counterintuitively it can be
+faster in addition to being vastly more memory efficient in the common case in
+analytics where we have table columns with many instances of the same string
+values.
+
+## Memory and Performance Benchmarks
+
+We can use the `memory_profiler` Python package to easily get process memory
```
Comment (xhochy) [inline site/_posts/2019-02-05-python-string-memory-0.12.md]:
```suggestion
We can use the [`memory_profiler`][2] Python package to easily get process memory
```

File: site/_posts/2019-02-05-python-string-memory-0.12.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+## Takeaways
+
+In Apache Arrow, our goal is to develop computational tools to operate natively
+on the cache- and SIMD-friendly efficient Arrow columnar format. In the
+meantime, though, we recognize that users have legacy applications using the
+native memory layout of pandas or other analytics tools. We will do our best to
+provide fast and memory-efficient interoperability with pandas and other
+popular libraries.
+
+[1]: https://www.slideshare.net/xhochy/extending-pandas-using-apache-arrow-and-numba
```
Comment (xhochy) [inline site/_posts/2019-02-05-python-string-memory-0.12.md]:
```suggestion
[1]: https://www.slideshare.net/xhochy/extending-pandas-using-apache-arrow-and-numba
[2]: https://pypi.org/project/memory-profiler/
```

**Review by xhochy — COMMENTED** [review]
Looks good, some small improvements but then this can go out tomorrow.

**Comment by wesm** [comment]
Oops sorry I missed these edits @xhochy

**Pull request closed without merging**